packages/sdk/client/src/client.ts

```diff
@@ -226,11 +209,30 @@ export class Client {
 
     log.trace('dxos.sdk.client.open', trace.begin({ id: this._instanceId }));
 
+    const { fromHost, fromIFrame } = await import('./services');
+    const { Monitor } = await import('./diagnostics');
+    const { EchoProxy, createDefaultModelFactory } = await import('./echo');
+    const { HaloProxy } = await import('./halo');
+    const { MeshProxy } = await import('./mesh');
+
+    this._config = this._options.config ?? new Config();
+    // NOTE: Must currently match the host.
+    this._modelFactory = this._options.modelFactory ?? createDefaultModelFactory();
+    this._services = await (this._options.services ?? (isNode() ? fromHost(this._config) : fromIFrame(this._config)));
```

Comment:
templated to roll these all up into a well-formed (readonly properties set in constructor) ClientRuntime class that we create/destroy and test separately. So that Client.initialize just becomes an async factory.

```ts
async initialize() {
  const { HaloProxy } = await import('./halo');
  _runtime = new ClientRuntime({ HaloProxy, ... });
}
...
checkInitialized() {
  if (!this._runtime) {
    throw
  }
}
...
async destroy() {
  await this._runtime.destroy();
  this._runtime = undefined;
}
...
createSpace() {
  // Public accessors since well defined.
  return this._runtime.echo.createSpace();
}
```